lib/Controller/SettingController.php

```diff
@@ -131,49 +132,48 @@ public function setAppValue($type, $value) {
 				$this->userManager->callForSeenUsers(function(IUser $user) {
-					$this->config->setUserValue($user->getUID(), 'facerecognition', 'recreate-clusters', 'true');
+					$this->settingService->setNeedRecreateClusters(true, $user->getUID());
 				});
 				break;
 			case 'memory-limits':
-				if (is_numeric ($value)) {
-					// Apply prundent limits.
-					if ($value < 1 * 1024 * 1024 * 1024) {
-						$value = 1 * 1024 * 1024 * 1024;
-						$message = $this->l10n->t("Recommended memory for analysis is at least 1GB of RAM.");
-						$status = self::STATE_ERROR;
-					} else if ($value > 4 * 1024 * 1024 * 1024) {
-						$value = 4 * 1024 * 1024 * 1024;
-						$message = $this->l10n->t("We are not recommending using more than 4GB of RAM memory, as we see little to no benefit.");
-						$status = self::STATE_ERROR;
-					}
-					// Valid according to RAM of php.ini setting.
-					$memory = MemoryLimits::getAvailableMemory();
-					if ($value > $memory) {
-						$value = $memory;
-						$message = $this->l10n->t("According to your system you can not use more than %s GB of RAM", ($value / 1024 / 1024 / 1024));
-						$status = self::STATE_ERROR;
-					}
-					// If any validation error saves the value
-					if ($status !== self::STATE_ERROR) {
-						$message = $this->l10n->t("The changes were saved. It will be taken into account in the next analysis.");
-						$this->config->setAppValue('facerecognition', $type, $value);
-					}
-				} else {
+				if (!is_numeric ($value)) {
 					$status = self::STATE_ERROR;
 					$message = $this->l10n->t("The format seems to be incorrect.");
 					$value = '-1';
 				}
+				// Apply prundent limits.
+				if ($value < SettingService::MINIMUM_MEMORY_LIMITS) {
+					$value = SettingService::MINIMUM_MEMORY_LIMITS;
+					$message = $this->l10n->t("Recommended memory for analysis is at least 1GB of RAM.");
+					$status = self::STATE_ERROR;
+				} else if ($value > SettingService::MAXIMUM_MEMORY_LIMITS) {
+					$value = SettingService::MAXIMUM_MEMORY_LIMITS;
+					$message = $this->l10n->t("We are not recommending using more than 4GB of RAM memory, as we see little to no benefit.");
+					$status = self::STATE_ERROR;
+				}
+				// Valid according to RAM of php.ini setting.
+				$memory = MemoryLimits::getAvailableMemory();
+				if ($value > $memory) {
+					$value = $memory;
+					$message = $this->l10n->t("According to your system you can not use more than %s GB of RAM", ($value / 1024 / 1024 / 1024));
+					$status = self::STATE_ERROR;
+				}
+				// If any validation error saves the value
+				if ($status !== self::STATE_ERROR) {
+					$message = $this->l10n->t("The changes were saved. It will be taken into account in the next analysis.");
+					$this->settingService->setMemoryLimits($value);
+				}
 				break;
 			case 'show-not-grouped':
-				$this->config->setAppValue('facerecognition', $type, $value);
+				$this->settingService->setShowNotGrouped($value == 'true' ? true : false);
```

Comment:
`===`?

Reply:
Yes .. I still have to fine tune some things. In particular, the jsons of the controllers sometimes return boolean as text, and I would prefer that they use true booleans. This forces to do some more conversions in the code, but finally it cleans the code quite a lot.